tests/material/bingham_test.cc

```diff
@@ -0,0 +1,565 @@
+    REQUIRE(stress(4) == Approx(0.000000e+00).epsilon(Tolerance));
+    REQUIRE(stress(5) == Approx(0.000000e+00).epsilon(Tolerance));
+  }
+}
+
+TEST_CASE("Bingham is checked in 3D", "[material][bingham][3D]") {
+  // Tolerance
+  const double Tolerance = 1.E-7;
+
+  const unsigned Dim = 3;
+
+  //! Check for id = 0
+  SECTION("Bingham id is zero") {
+    unsigned id = 0;
+    auto material = Factory<mpm::Material<Dim>, unsigned>::instance()->create(
+        "Bingham3D", std::move(id));
+    REQUIRE(material->id() == 0);
+  }
+
+  SECTION("Bingham id is positive") {
+    //! Check for id is a positive value
+    unsigned id = std::numeric_limits<unsigned>::max();
+    auto material = Factory<mpm::Material<Dim>, unsigned>::instance()->create(
+        "Bingham3D", std::move(id));
+    REQUIRE(material->id() == std::numeric_limits<unsigned>::max());
+  }
+
+  //! Read material properties
+  SECTION("Bingham check properties") {
+    unsigned id = 0;
+    auto material = Factory<mpm::Material<Dim>, unsigned>::instance()->create(
+        "Bingham3D", std::move(id));
+    REQUIRE(material->id() == 0);
+
+    // Initialise material
+    Json jmaterial;
+    jmaterial["density"] = 1000.;
+    jmaterial["youngs_modulus"] = 1.0E+7;
+    jmaterial["poisson_ratio"] = 0.3;
+    jmaterial["tau0"] = 771.8;
+    jmaterial["mu"] = 0.0451;
+    jmaterial["critical_shear_rate"] = 0.2;
+
+    // Check material status before assigning material property
+    REQUIRE(material->status() == false);
+
+    // Get material properties
+    REQUIRE(material->property("density") ==
+            Approx(std::numeric_limits<double>::max()).epsilon(Tolerance));
+
+    // Check for property that does not exist
+    REQUIRE(material->property("noproperty") ==
+            Approx(std::numeric_limits<double>::max()).epsilon(Tolerance));
+
+    material->properties(jmaterial);
+
+    // Check material status after assigning material property
+    REQUIRE(material->status() == true);
+
+    // Get material properties
+    REQUIRE(material->property("density") ==
+            Approx(jmaterial["density"]).epsilon(Tolerance));
+  }
+
+  SECTION("Bingham check stresses with no strain rate") {
+    unsigned id = 0;
+    auto material = Factory<mpm::Material<Dim>, unsigned>::instance()->create(
+        "Bingham3D", std::move(id));
+    REQUIRE(material->id() == 0);
+
+    // Initialise material
+    Json jmaterial;
+    jmaterial["density"] = 1000.;
+    jmaterial["youngs_modulus"] = 1.0E+7;
+    jmaterial["poisson_ratio"] = 0.3;
+    jmaterial["tau0"] = 771.8;
+    jmaterial["mu"] = 0.0451;
+    jmaterial["critical_shear_rate"] = 0.2;
+
+    material->properties(jmaterial);
+
+    // Check material status after assigning material property
+    REQUIRE(material->property_handle() == true);
+
+    // Initialise stress
+    mpm::Material<Dim>::Vector6d stress;
+    stress.setZero();
+    REQUIRE(stress(0) == Approx(0.).epsilon(Tolerance));
```

Comment:
Remove these tests of stress, of course it should be zero.

Reply:
Is it always zero when you initialize it? I thought we need to set zero to make it zero according to this: https://eigen.tuxfamily.org/dox/group__TutorialAdvancedInitialization.html